presto-sqlserver/src/main/java/io/prestosql/plugin/sqlserver/SqlServerTableProperties.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.sqlserver;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.plugin.jdbc.TablePropertiesProvider;
+import io.prestosql.spi.session.PropertyMetadata;
+
+import java.util.List;
+import java.util.Map;
+
+import static io.prestosql.plugin.sqlserver.DataCompression.NONE;
+import static io.prestosql.spi.session.PropertyMetadata.enumProperty;
+
+public class SqlServerTableProperties
+        implements TablePropertiesProvider
+{
+    public static final String DATA_COMPRESSION = "data_compression";
+
+    private final List<PropertyMetadata<?>> tableProperties = ImmutableList.of(
+            enumProperty(
+                    DATA_COMPRESSION,
+                    "DataCompression type for table",
+                    DataCompression.class,
+                    NONE,
+                    false));
+
+    @Override
+    public List<PropertyMetadata<?>> getTableProperties()
+    {
+        return tableProperties;
+    }
+
+    public static DataCompression getDataCompression(Map<String, Object> tableProperties)
```

Comment:
Shouldn't you return `Optional` here?

Reply:
We passed a default value to `enumProperty` so it should be non-null I guess

Reply:
So I guess default should be `null` (unknown, don't care (default when you create a table)). 

We should return `NONE` if we known that all table partitions are not compressed.